lib/oga/entity_decoder.rb

```diff
@@ -13,6 +13,7 @@ def self.try_decode(input, html = false)
     # @return [String]
     #
     def self.decode(input, html = false)
+      fail ArgumentError, "decode was sent nil" if input.nil?
```

Comment:
Also, if this were needed it would still need some changes:

* Use `raise` instead of `fail`. I'm aware tools such as Rubocop advocate `fail` for byzantine reasons, but Oga uses `raise` everywhere and I like to keep it that way.
* The error message itself is not particularly useful.
* Changes such as these _require_ specs to be added to cover the new behaviour, these are currently missing.